src/Elastic.Apm.AspNetFullFramework/ApplicationConfigurationReader.cs

```diff
@@ -7,9 +9,11 @@
 
 namespace Elastic.Apm.AspNetFullFramework
 {
-	internal class FullFrameworkConfigReader : EnvironmentConfigurationReader
+	internal class ApplicationConfigurationReader : EnvironmentConfigurationReader
```

Comment:
I also don't really see why we should rename this.

Reply:
`Elastic.Apm.AspNetCore` and `Elastic.Apm.AspNetFullFramework` are corresponding projects, one for .NET Core, one for Full Framework. Therefore, any corresponding classes should follow a similar pattern.

- `MicrosoftExtensionsConfig` extends `AbstractConfigurationReader` and realises `IConfigurationReader`.
- `FullFrameworkConfigReader` extends `EnvironmentConfigurationReader`.

Since `MicrosoftExtensionsConfig` was renamed more appropriately as explained in [my reply](https://github.com/elastic/apm-agent-dotnet/pull/309/files#r300779132) to your other comment, `FullFrameworkConfigReader` should follow a similar pattern. Including `FullFramework` in the name makes little sense as it is already in the Full Framework project and therefore in the `AspNetFullFramework` namespace. Avoid repetition in namespaces and class names ensures DRY is followed. It is therefore renamed the same as `MicrosoftExtensionsConfig`. Both are `ApplicationConfigurationReader` as this indicates what they do in their respective projects. They'll never conflict because they're in independent projects with different namespaces.

**Advantage:** The reader can understand what the class does and will be able to see the correspondence between the .NET Core and Full Framework projects much more easily.

In any case, I've reverted the change and will put it in a separate PR.